### DISCUSSION TITLE
博士论文里想放一些源代码，学校有没有格式要求？

### DISCUSSION THREAD

**Discussion by YanQH-Gausoul** [tuna/thuthesis#740] [Q&A] [answered]
我有些mathematica的代码想放在博士论文的附录里，但是学校的 word 模板里或者 thuthesis 的手册里都没有提到代码环境，有没有什么可以参考的规范或者格式要求？
手册里有提到 minted 宏包，但是用起来好像比较麻烦。 listings 宏包简单一些，但是没找到任何规范可以参考。是不是可以把代码看成图片之类的来处理？这样图片本身啥样不重要？

**Comment by zggl** [comment]
没说什么语言，基本都是从ctan下载文档，照着文档处理一下就出来了。现在我建议用mcode.sty，matlab给的一个代码包。

On Wed, Mar 16, 2022 at 10:43 AM YanQH-Gausoul ***@***.***>
wrote:

> 我有些mathematica的代码想放在博士论文的附录里，但是学校的 word 模板里或者 thuthesis
> 的手册里都没有提到代码环境，有没有什么可以参考的规范或者格式要求？
> 手册里有提到 minted 宏包，但是用起来好像比较麻烦。 listings 宏包简单一些，但是没找到任何规范可以参考。
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tuna/thuthesis/discussions/740>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD66RNLFLH7QQS7QSTAFNC3VAFDDFANCNFSM5Q2QQJCA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
      祝好！

赵国亮
========================================================

通信地址：1. 呼和浩特 大学西路235号 内蒙古大学 卓越楼913
               PHONE:18604604983

E-Mail ： ***@***.******@***.***
========================================================

**Comment by Harry-Chen** [comment, marked as answer]
`minted` 比较现代，支持的语言更多，功能更丰富，使用也没有显著比 `lstlisting` 更麻烦。

**Reply by YanQH-Gausoul** [reply]
我重写了一下问题，主要还是想问有没有格式要求

**Reply by Harry-Chen** [reply]
学校的写作指南里没有相关的格式要求，根据以往的经验，也是应该可以自行处理的。

**Comment by Auyuir** [comment]
https://blog.csdn.net/Stool_Monster/article/details/107610030
这个博客对pygmentize问题实测有效。
